zebra/zebra_mpls.c

```diff
@@ -313,14 +308,8 @@ static int lsp_uninstall(struct zebra_vrf *zvrf, mpls_label_t label)
 		if (lsp_processq_add(lsp))
 			return -1;
 	} else if (!lsp->nhlfe_list
-		   && !CHECK_FLAG(lsp->flags, LSP_FLAG_SCHEDULED)) {
-		if (IS_ZEBRA_DEBUG_MPLS)
-			zlog_debug("Del LSP in-label %u flags 0x%x",
-				   lsp->ile.in_label, lsp->flags);
-
-		lsp = hash_release(lsp_table, &lsp->ile);
-		XFREE(MTYPE_LSP, lsp);
-	}
+		   && !CHECK_FLAG(lsp->flags, LSP_FLAG_SCHEDULED))
+		lsp_dtor(lsp_table, &lsp);
```

Comment:
I realize the creation doesn't follow our normal naming standards but `XXX_dtor` is not what we call deletion of memory within FRR.  Please align.

Reply:
ok, I've pushed a rename to `lsp_free()`